tiledb/sm/array/array_directory.cc

```diff
@@ -98,80 +112,129 @@ Status ArrayDirectory::load() {
   assert(!loaded_);
 
   std::vector<ThreadPool::Task> tasks;
-  std::vector<URI> fragment_uris_v1_v11;
-  std::vector<URI> fragment_uris_v12_or_higher;
-  URI latest_fragment_meta_uri_v1_v11;
+  std::vector<URI> root_dir_uris;
+  std::vector<URI> commits_dir_uris;
   URI latest_fragment_meta_uri_v12_or_higher;
 
-  // Load (in parallel) the root directory data
-  if (!only_schemas_)
+  if (mode_ != ArrayDirectoryMode::SCHEMA_ONLY) {
```

Comment:
For better clarity now that we have multiple modes, is it possible to make cases based on the mode that is true rather than the mode that is not true? Can we also break this into different functions, e.g., `load_schema_only`, `load_commits`, `load_<mode>`, etc? I understand that there may be a bit of overlap, but it will be easier to read and maintain down the road, especially if we add more modes.

Reply:
I was unable to get to this one for now. Maybe we can file a follow up?